packages/common/react-components/src/components/DensityProvider/DensityProvider.tsx

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { createContext, PropsWithChildren } from 'react';
+
+import { Density } from '../../props';
+
+export interface DensityContextValue {
+  density?: Density;
+}
+
+export type DensityProviderProps = PropsWithChildren<{
+  density?: Density;
+}>;
+
+export const DensityContext = createContext<DensityContextValue>({ density: 'coarse' });
+
+export const DensityProvider = ({ density, children }: DensityProviderProps) => (
+  <DensityContext.Provider value={{ density }}>{children}</DensityContext.Provider>
```

Comment:
While this technically is a provider from the react framework POV, to a user, this is a container so perhaps there's another abstraction like `<ControlGroup>` where one might specify density and elevation, etc. to influence a more horizontal provider (rather than multiple).

```tsx
<Group density='coarse' elevation={2} variant='ghost'>
  <Button>1</Button>
  <Button>2</Button>
  <Button variant='outline'>3</Button>
</Group>
```

Reply:
Sure — I don’t anticipate many will want to use `DensityProvider` directly. `List` and others already use it internally to communicate density to their children, and I’d imagine other grouping components could offer `density` as a prop and do the same like you illustrate here.